GeoUtils/vector_tools.py

```diff
@@ -1,12 +1,124 @@
 """
 GeoUtils.vector_tools provides a toolset for working with vector data.
 """
+import warnings
+import numpy as np
 import geopandas as gpd
+import rasterio as rio
+from rasterio import warp, features
+
+from GeoUtils.raster_tools import Raster
 
 
 class Vector():
     """
     Create a Vector object from a fiona-supported vector dataset.
     """
-    def __init__(self):
-        pass
+
+    def __init__(self, filename):
+        """
+        Load a fiona-supported dataset, given a filename.
+
+        :param filename: The filename of the dataset.
+        :type filename: str
+
+        :return: A Vector object
+        """
+
+        ds = gpd.read_file(filename)
+        self.ds = ds
+        self.name = filename
+
+    def crop2raster(self, rst):
+        """
+        Update self so that features outside the extent of a raster file are cropped. Reprojection is done on the fly if both data set have different projections.
+
+        :param rst: A Raster object or string to filename
+        :type rst: Raster object or str
+        """
+        # If input is string, open as Raster
+        if isinstance(rst, str):
+            rst = Raster(rst)
+
+        # Convert raster extent into self CRS
+        # Note: could skip this if we could test if rojections are same
+        # Note: should include a method in Raster to get extent in other projections, not only using corners
+        left, bottom, right, top = rst.bounds
+        x1, x2, y1, y2 = warp.transform_bounds(
+            rst.crs, self.ds.crs, left, bottom, right, top)
+        self.ds = self.ds.cx[x1:x2, y1:y2]
+
+    def create_mask(self, rst=None, crs=None, xres=None, yres=None, bounds=None, in_value=255, out_value=0):
+        """
+        Crop a vector file to the extent of a raster file. Reprojection is done on the fly if both data set have different projections.
```

Comment:
I don't think that the headline of this docstring accurately defines what the function actually does?

How about 

"Rasterize the vector features into a raster which has the extent of the provided raster file. 

Vector features which fall outside the bounds of the raster file are not written to the new mask file.